### PR TITLE
feat: adds support for ed25199 and secp256k1

### LINKED ISSUE #30: Add ed25519 and secp256k1
I saw that it only supports _rsa_ keys here: https://github.com/libp2p/js-libp2p-keychain/blob/master/src/keychain.js#L201

But the underlying library (https://github.com/libp2p/js-libp2p-crypto) also supports ed25519, and also secp256k1 if you add https://github.com/libp2p/js-libp2p-crypto-secp256k1.

Is there anything holding this feature back or is it something simple that I could create a PR for it so it will be supported downstream in js-ipfs?

### PR DESCRIPTION
**Pull request by AlbertoElias**
Fixes #30 

This is blocked by: https://github.com/libp2p/js-libp2p-crypto-secp256k1/pull/13

@vasco-santos Could you review?

### CODE REVIEW

**Comment by AlbertoElias** [comment]
@vasco-santos done!

**Comment by AlbertoElias** [comment]
@vasco-santos That should be done now. I also updated exporting and importing to fix marshaling.

File: src/keychain.js [hunk 1]
```diff
@@ -376,65 +373,64 @@ class Keychain {
     if (!validateKeyName(name)) {
```
Comment (vasco-santos) [inline src/keychain.js]: Make the password optional, that is, verify initially on both `exportKey` and `importKey` if password is a function and callback does not exist. If so, assign the callback to the password and password to undefined. It is cleaner to omit the password than having to pass a null parameter. Also add `(optional)` in the js docs please

**Comment by AlbertoElias** [comment]
@vasco-santos Everything is fixed, but it seems that what needs updating is https://github.com/libp2p/js-libp2p-crypto/pull/137 I've tampered my package-lock.json to get everything running, but if `js-libp2p-crypto` decides to do a minor release instead of patch, `js-libp2p-keychain`'s package.json would need to be updated as well

File: src/keychain.js (hunk 1 above)
Comment (AlbertoElias) [reply src/keychain.js]: Done!

**Comment by raullaprida** [comment]
I'm creating a project using js libp2p and I created a Keystore where I store secp25k1 keys using PEM encrypted PKCS #8.  (so now password wouldn't be just for RSA keys).  Tests include being able to use OpenSSL to verify the encrypted keys.
Would that be a desired feature for the keychain? (I don't want to make a PR and be automatically rejected)

**Pull request merged**